packages/website/components/Playground/useUrlSync.tsx

```diff
@@ -1,20 +1,24 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import { useRouter } from 'next/router';
 import { useDebounce } from 'use-debounce';
 import { useActiveCode } from '@codesandbox/sandpack-react';
 import * as coder from '../../utils/coder';
 
 export function useUrlSync() {
   const router = useRouter();
+  const [codeUrl, setCodeUrl] = useState('');
   const { code } = useActiveCode();
   const [debouncedCode] = useDebounce(code, 1000);
 
   useEffect(() => {
     const code = coder.encode(debouncedCode);
-    if (code === router.query.code) {
-      return;
+
+    if (code !== router.query.code) {
+      const href = `/playground?code=${code}`;
+      window.history.replaceState('', '', href);
+      setCodeUrl(window.location.origin + href);
     }
-    const href = `/playground?code=${code}`;
-    router.push(href, undefined, { shallow: true });
   }, [debouncedCode]);
+
+  return { router, codeUrl };
```

Comment:
Since you don't use the `router` returned here, can't it be removed?

Reply:
good catch!